packages/stateless/src/lib/provider/provider.interface.ts

```diff
@@ -56,14 +56,36 @@ export interface ISendMessageSuccessResponse {
   date?: string;
 }
 
+export enum EmailEventStatusEnum {
+  OPEN = 'open',
+  DELIVERY = 'delivery',
+  BOUNCE = 'bounce',
+  DROPPED = 'dropped',
+}
+
+export interface IEventBody {
+  status: string;
+  date: string;
+  attempts?: number;
+  response?: string;
+  externalId?: string;
+}
+
+export interface IEmailEventBody extends IEventBody {
```

Comment:
@ainouzgali and @scopsy what do you think about storing this data only, some of the data we get back are data we already have or is data that not everyone send us. Sendgrid is one of those that sends least data...